server/routes/appointmentAPI.js

```diff
@@ -25,17 +26,61 @@ router.post("/api/appointment", async (req, res) => {
   }
 });
 
-// GET all appointments for user. Query via userId
-router.get("/api/all-appointments", auth, (req, res) => {
-  db.Appointment.find({ hostUserId: req.userId })
-    .then((data) => {
-      //data: [{_id, email, eventId, hostUserId, name, time, timezone},...]
-      res.send(data);
-    })
-    .catch((error) => {
-      console.log(error);
-      res.status(500).send(error);
-    });
+// GET all appointments for user. Query via userId.
+// Returns an array (could be of length 0) containing the details of the appointment
+// [{
+//   _id, // appointmentId
+//   eventId,
+//   hostUserId,
+//   name, // attendee's name
+//   email, // attendee's email
+//   time, // ISO8601 foramtted date and time
+//   timezone,
+//   duration, // a number in minute
+//   description,
+//   color, // the color of the flag
+//   link, // the incomplete link for building invitation link: <hostName>/<eventName>
+//   members,
+//   active, // indicate whether the event is active or not
+//   eventName
+// }, ...]
+router.get("/api/all-appointments", auth, async (req, res) => {
+  try {
+    const appointments = await db.Appointment.find({ hostUserId: req.userId });
+    if (!appointments.length) {
+      throw new Error("noAppointmentsFound");
+    }
+
+    const appointmentsWithEventDetailsPromise = appointments.map(
+      async (appointment) => {
+        const eventDetails = await getEventDetailViaId(appointment.eventId);
+        // Order matters, both of them contain a `name` key,
+        // in eventDetails, that's the event's name,
+        // in appointment, that's the attendee's name
+        // I will save only the attendee's name
+        // Also, mongoose query returned object is not a regular object
+        // need to convert before spreading
+        // big pain

```

Comment:
FYI large comments like this explaining approach are often frowned upon. Not really worth changing but for the future when you feel a comment is necessary try to either get the point across with variable names or if that's not possible then try to keep the comment to one line like `// spread eventDetails first because of duplicate keys`